service/src/test/java/bio/terra/cbas/controllers/TestRunSetsApiController.java

```diff
@@ -0,0 +1,158 @@
+  @MockBean private RunDao runDao;
+
+  // These beans are not intended to be hit by these tests, so it feels safe (even
+  // beneficial?) to leave them in the context as empty mocks:
+  @MockBean private PublicApiController publicApiController;
+  @MockBean private RunsApiController runsApiController;
+  @MockBean private BatchApiController batchApiController;
+  @MockBean private SwaggerController swaggerController;
+  @MockBean private CromwellClient cromwellClient;
+  @MockBean private WdsClient wdsClient;
+
+  // This mockMVC is what we use to test API requests and responses:
+  @Autowired private MockMvc mockMvc;
+
+  // The object mapper is pulled from the BeanConfig and used to convert to and from JSON in the
+  // tests:
+  @Autowired private ObjectMapper objectMapper;
+
+  @Test
+  void chainCallsTogether() throws Exception {
+
+    final String workflowUrl = "www.example.com/wdls/helloworld.wdl";
+    final String entityType = "MY_ENTITY_TYPE";
+    final String entityId = "MY_ENTITY_ID";
+    final String entityAttribute = "MY_ENTITY_ATTRIBUTE";
+    final int entityAttributeValue = 100;
+    EntityAttributes entityAttributes = new EntityAttributes();
+    entityAttributes.put(entityAttribute, entityAttributeValue);
+    final String cromwellWorkflowId = UUID.randomUUID().toString();
+    final String cromwellWorkflowStatus = "Submitted";
+
+    // Set up API responses:
+    when(wdsService.getEntity(entityType, entityId))
+        .thenReturn(
+            new EntityResponse().type(entityType).id(entityId).attributes(entityAttributes));
+
+    ArgumentCaptor<Map<String, Object>> inputsCaptor = ArgumentCaptor.forClass(Map.class);
+    when(cromwellService.submitWorkflow(eq(workflowUrl), inputsCaptor.capture()))
+        .thenReturn(
+            new WorkflowIdAndStatus().id(cromwellWorkflowId).status(cromwellWorkflowStatus));
+
+    RunSetRequest request =
+        new RunSetRequest()
+            .workflowUrl(workflowUrl)
+            .addWorkflowParamDefinitionsItem(
+                new WorkflowParamDefinition()
+                    .parameterName("myworkflow.mycall.inputname1")
+                    .parameterType(ParameterValueType.STRING)
+                    .source(
+                        new ParameterDefinitionLiteralValue()
+                            .parameterValue("literal value")
+                            .type(ParameterDefinition.TypeEnum.LITERAL)))
+            .addWorkflowParamDefinitionsItem(
+                new WorkflowParamDefinition()
+                    .parameterName("myworkflow.mycall.inputname2")
+                    .parameterType(ParameterValueType.INT)
+                    .source(
+                        new ParameterDefinitionEntityLookup()
+                            .entityAttribute("MY_ENTITY_ATTRIBUTE")
+                            .type(ParameterDefinition.TypeEnum.ENTITY_LOOKUP)))
+            .wdsEntities(new WdsEntitySet().entityType(entityType).addEntityIdsItem(entityId));
```

Comment:
It depends on what you're intending to test in this test, but I would write this using the JSON data itself instead of using the request types. There are pros and cons either way of course.. if the request format changes then the JSON is no longer valid and will need to be updated. Using a text block and `formatted()` makes this pretty easy to do:
```suggestion
    String requestContent = """
        {
          "workflow_url" : "%s",
          "workflow_param_definitions" : [ {
            "parameter_name" : "myworkflow.mycall.inputname1",
            "parameter_type" : "String",
            "source" : {
              "type" : "literal",
              "parameter_value" : "literal value"
            }
          }, {
            "parameter_name" : "myworkflow.mycall.inputname2",
            "parameter_type" : "Int",
            "source" : {
              "type" : "entity_lookup",
              "entity_attribute" : "MY_ENTITY_ATTRIBUTE"
            }
          } ],
          "wds_entities" : {
            "entity_type" : "%s",
            "entity_ids" : [ "%s" ]
          }
        }
        """.formatted(workflowUrl, entityType, entityId);
```